### ISSUE TITLE
Changes compared to ‘feathers-pinia‘

### ISSUE THREAD

**Issue by fratzinger** [artesa/feathers-pinia#1]
### BaseModel:
- Constructor not accepting any arguments. Leave any defaults or anything to the inherited classes
  - `ModelDefineOptions` removed. `__isClone` as class property predefined, not with `defineProperty`
  - all Models are `__isClone: false` by default. Simply make it `true` in `store.clone(...)`
- Not calling `instanceDefaults` by default, use `this.init(data)` instead
  - `constructor(data)` with `instanceDefaults(data)` and `init(data)` has the problem, that `data` always is on top and therefore need to be merged last. `instanceDefaults` was called in `BaseModel`. Calling `this.init` would call `instanceDefaults` two times.

### clientAlias / models / registerClient
- completely removed

### Chore
- removed a bunch of unused packages

**Comment by marshallswain** [comment]
I think these are good calls.  We might want to keep using `defineProperty` for `__isClone` to make it non-iterable (so it won't be sent to the API).  Alternatively we keep showing people how to remove the attribute in a before-all hook on the client.

**Comment by marshallswain** [comment]
> Not calling instanceDefaults by default, use this.init(data) instead

This is the exact direction I wanted to take the constructor. :+1:

**Comment by fratzinger** [comment]
Thanks for reviewing and your comments! And thanks for your open mind on these changes! Highly appreciated.

> We might want to keep using `defineProperty` for `__isClone` to make it non-iterable (so it won't be sent to the API).

Good point! I forgot that `defineProperty` is `enumerable: false` by default. Because it was explicitly in feathers-vuex. As of now, I filter it through `cleanData` [see source](https://github.com/Artesa/feathers-pinia/blob/v1-setup-store/src/service/use-service.ts#L442)

I'll change it to `defineProperty`. That makes so much sense. Also for serialization. But the point stays, I think: don't set `__isClone` conditionally in the constructor depending on the options but make it `false` by default.

**Comment by marshallswain** [comment]
> As of now, I filter it through cleanData

Yes, this is how I was doing it until a few months ago.  I appreciate not having the extra setup step if we can avoid it.

> Also for serialization. But the point stays, I think: don't set __isClone conditionally in the constructor depending on the options but make it false by default.

Good plan
